tests/end2end/playwright/pages/project.js

```diff
@@ -2,6 +2,17 @@
 import {expect, Locator, Page} from '@playwright/test';
 import { gotoMap } from '../globals';
 
+/**
+ * @typedef {Object} logMessage
+ * @property {string} type - the log type :One of the following values:
+ * 'log', 'debug', 'info', 'error', 'warning', 'dir', 'dirxml', 'table',
+ * 'trace', 'clear', 'startGroup', 'startGroupCollapsed', 'endGroup',
+ * 'assert', 'profile', 'profileEnd', 'count', 'timeEnd'.
+ * @property {string} message - the log message text
+ * @property {string} location - the log message location in one line
+ * @see https://playwright.dev/docs/api/class-consolemessage
+ */
```

Comment:
Sorry, I'm not sure to understand the purpose of this block ?

Reply:
It is done to describe `logMessage` object to simplify `logs`  property description.

https://jsdoc.app/tags-typedef